### PR TITLE
Semikolon in Whitelist + Add new boolean property: BtchBookg

### PR DESCRIPTION
**Pull request by MicrotronX**
plus unittests

### CODE REVIEW

File: SEPACreditTransfer.pas [hunk 1]
```diff
@@ -83,6 +83,7 @@   TCreditTransferTransactionInformation = class
   TCreditTransferPaymentInformation = class
   private
     fPmtInfId: String;                                 // payment information identification
+    fBtchBookg:boolean;                                // Batchbooking parameter (at Sepa level default true if not defined)
```
Comment (aspettl) [inline SEPACreditTransfer.pas]: Please format the same as the other lines (whitespace before type, type with capital letter) - also in the other file, of course.

File: SEPACreditTransfer.pas [hunk 2]
```diff
@@ -271,6 +273,8 @@ constructor TCreditTransferPaymentInformation.Create;
   fChrgBr           := CHRG_BR_SLEV;
   fDbtrAcct         := TAccountIdentification.Create;
   fDbtrAgt          := TFinancialInstitution.Create;
+
+  fBtchBookg    :=  false; // Set it default to false, we want to have details
```
Comment (aspettl) [inline SEPACreditTransfer.pas]: I do not agree on the default. According to the documentation, "true" is the default and for details one needs a special agreement. We cannot assume here that this is what the user has and wants.

File: SEPACreditTransfer.pas [hunk 3]
```diff
@@ -373,11 +377,18 @@ function TCreditTransferPaymentInformation.Validate(const schema: String; const
 procedure TCreditTransferPaymentInformation.SaveToStream(const stream: TStream; const schema: String);
 var
   i: Integer;
+    function _Boolean2Xml(_value:boolean):string;
+    begin
+        if _value then result:='true' else result:='false';
+    end;
```
Comment (aspettl) [inline SEPACreditTransfer.pas]: This function is defined twice in this PR. Please extract to the "SEPACommon.pas" file (and adapt indentation and whitespaces to the used code style, please also remove the "_" prefix in the variable name).

File: SEPACommon.pas [hunk 4]
```diff
@@ -245,6 +245,9 @@ function CharIsSEPAWhitelisted(const c: Char): Boolean;
             (c = '''') or (c = ':') or (c = '?') or
             (c = ',') or (c = '-') or (c = ' ') or
             (c = '(') or (c = '+') or (c = '.') or
+
+            (c = ';') or   // We allow the ; char so we can use XML-Escapes like &amp;
```
Comment (aspettl) [inline SEPACommon.pas:250]: I'm still not sure this is the right way. On the other hand, it gives the flexibility to use escaping when the own financial institution accepts this.
In the future: please prepare separate branches/pull requests for different features.

File: README.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Delphi-Unit zur Erstellung von SEPA-XML-Dateien
+# Delphi-Unit zur Erstellung von SEPA-XML - Dateien
```
Comment (aspettl) [inline README.md]: Please revert, I don't see any reason for this change.

**Review by aspettl — CHANGES_REQUESTED** [review]
Thanks for the pull request! I have added a few comments, would be great if you can implement these changes.

**Comment by MicrotronX** [comment]
Hello Aaron,

wir haben soeben alle deine vorgeschlagenen Änderungen durchgeführt. Die
Readme war nur ein Test auf unserer Seite ob es überhaupt mit dem Fork /
Pullrequest funktioniert (ist das erste mal ;-)
Werde gleich einen neuen Pull-Request erstellen.

Mfg. Yusuf Zorlu

2018-08-04 13:47 GMT+02:00 Aaron Spettl <notifications@github.com>:

File: SEPACommon.pas (hunk 4 above)
Comment (MicrotronX) [reply SEPACommon.pas:250]: We don't have problems with xml-escapes with customers in germany, austria and luxembourg. At the moment this is working without problems. Without escaping all customers had problems with special chars.

**Comment by aspettl** [comment]
Danke!

**Pull request merged**